flang/lib/Lower/OpenMP.cpp

```diff
@@ -716,6 +727,62 @@ genOMP(Fortran::lower::AbstractConverter &converter,
   createBodyOfOp<omp::CriticalOp>(criticalOp, converter, currentLocation, eval);
 }
 
+ void Fortran::lower::genOpenMPSectionsBlock(Fortran::lower::AbstractConverter &converter, Fortran::lower::pft::Evaluation &eval){
+     auto &firOpBuilder = converter.getFirOpBuilder(); 
+     auto currentLocation = converter.getCurrentLocation();
+     mlir::omp::SectionOp sectionOp = firOpBuilder.create<mlir::omp::SectionOp>(currentLocation);
+     createBodyOfOp<omp::SectionOp>(sectionOp, converter, currentLocation, eval);
+ }
+
+ // TODO: Add support for reduction
+ static void
+ genOMP(Fortran::lower::AbstractConverter &converter,
+        Fortran::lower::pft::Evaluation &eval,
+        const Fortran::parser::OpenMPSectionsConstruct &sectionsConstruct) {
+     auto &firOpBuilder = converter.getFirOpBuilder();
+     auto currentLocation = converter.getCurrentLocation();
+     SmallVector<Value> privateClauseOperands, firstPrivateClauseOperands,
+         lastPrivateClauseOperands, reductionVars, allocateOperands,
```

Comment:
Update: Maybe giving an error and stopping compilation is better to avoid unwanted behavior. (This was suggested for the sections lowering patch by reviewer for me, and I think that makes more sense than silently generating something else or emitting a warning).

Reply:
Ok thanks. I guess you would be referring to the review comments on https://reviews.llvm.org/D115030

Reply:
Yes. I think errors should be shown in others too (perhaps as a separate patch for the ease of review maybe?), as we do not want to have unwanted behavior by the binary when the user especially meant it to behave differently.